### ISSUE TITLE
In screensaver (but not channel), metadata is loading - but no images

### ISSUE THREAD

**Issue by darryllee** [chtaylo2/Roku-GooglePhotos#390]
Hi - in both preview and after the timeout, the screensaver "loads" with metadata listing different photo names, but the image is black.

I've tried lower screen resolutions, as well as switching from a Week in History to a "standard" album.

I even tried uninstalling and reinstalling the channel, and in preview, it's showing "Screensaver Paused" (one of your default images?) with no image.

This is on a Roku Streambar (OS version 10.0.0 build 4209-95), where it was working great up until maybe a week ago.

OH, and I did test going to the Channel directly and images **did** properly load up there. Odd!

**Comment by darryllee** [comment]
Bah. It was my Pihole. Weird. I was looking to see if any queries were being blocked before and didn't see any.

Bummer. I love blocking ads on my Roku...

**Comment by darryllee** [comment]
Odd. When I do Preview Screensaver, I see my Roku trying to go to www.googleapis.com, photolibrary.googleapis.com, and lh3.googleusercontent.com, and according to the Pi-hole, it successfully either returns a cached result, or passes those requests through to my upstream DNS server (I'm using familyshield.opendns.com with the Pi-hole.)

What I'm **not** seeing are any requests being blocked.

**Comment by darryllee** [comment]
And, after disabling Pi-hole for the past 5 hours, the problem is back - metadata loads with date/filename of images, but black screen displayed. So it's not a Pi-hole problem. Any ideas?

BTW, *love* the app. My kids aren't as excited about seeing pictures of themselves anymore, but my wife and I still love it. A similar feature was one of my favorite things about Google's Chromecasts.

**Comment by darryllee** [comment]
Ooooh, could this be a quota issue? Is there any way to configure the tool so that I could drop in my own API key to avoid hitting those limits?

**Comment by chtaylo2** [comment]
You’re correct, it’s due to a quota issue which often happens late in the day.  It resets at 12:00 AM PT. It’s not an issue with default Pi-hole installs.

Allowing developer API keys is a possible solution, but making it a reality is complex. Due to the registration only allowed from web redirects, what https://www.photoviewapp.com does, I’d have to code this into that site. Not sure how many people would actually use it..

**Comment by toddbutters** [comment]
I would be interested in using my own API keys.
Could those interested host their own version of the photoviewapp website with their own API keys and link their Roku using it?

**Comment by darryllee** [comment]
A developer whose app also uses the Google Photos API ran into a similar issue, and he ended up joining the Google Photos Partner Program to increase quota limits:

https://developers.google.com/photos/partner-program/overview

He tells me that they were very helpful and responsive.

**Comment by chtaylo2** [comment]
@darryllee  - Yes the app is part of the Partner Program. In fact, we were one of the early adopters. Daily this channel handles 10's of millions of photos across all users. We're just still limited by this amount.   Thanks for using the app
